test/csuite/wt8659_reconstruct_database_from_logs/main.c

```diff
@@ -0,0 +1,379 @@
+         * Only do this if the copy process was entirely successful.
+         */
+        testutil_check(session->truncate(session, "log:", cursor, NULL, NULL));
+    }
+
+    testutil_check(cursor->close(cursor));
+}
+
+/*
+ * prepare_folders --
+ *     Prepare all working folders required for the test.
+ */
+static void
+prepare_folders(void)
+{
+    reset_dir(home_live);
+    reset_dir(home_full);
+    reset_dir(home_incr);
+    reset_dir(home_incr_copy);
+}
+
+/*
+ * cleanup --
+ *     Test's cleanup.
+ */
+static void
+cleanup(void)
+{
+    testutil_check(conn->close(conn, NULL));
+
+    remove_dir(home_full);
+    remove_dir(home_incr);
+    remove_dir(home_live);
+    remove_dir(home_incr_copy);
+}
+
+/*
+ * reopen_conn --
+ *     Close and reopen connection to the database.
+ */
+static void
+reopen_conn(void)
+{
+    if (conn != NULL) {
+        printf("Reopening connection\n");
+        testutil_check(conn->close(conn, NULL));
+        conn = NULL;
+        session = NULL;
+    }
+
+    testutil_check(wiredtiger_open(home_live, NULL, conn_config, &conn));
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+}
+
+/*
+ * validate --
+ *     Validate the database against incremental backup. To do that we need to take a full backup of
+ *     the database. Also we have to make a copy of the incremental backup to avoid "Incremental
+ *     backup after running recovery is not allowed" error.
+ */
+static void
+validate(bool after_reconnect)
+{
+    /*
+     * The full backup here is only needed for testing and comparison purposes. A normal incremental
+     * backup procedure would not include this.
+     */
+    printf("Taking full backup\n");
+    take_full_backup(home_live, home_full);
+
+    /*
+     * Taking the incremental backup also calls truncate to archive the log files, if the copies
+     * were successful. See that function for details on that call. The truncation only happens
+     * after we reconnected to the database.
+     */
+    printf("Taking incremental backup\n");
+    take_incr_backup(home_incr, after_reconnect);
+
+    /*
+     * Dump tables from the full backup and incremental backup databases, and compare the dumps.
+     */
+    printf("Dumping and comparing data\n");
+    testutil_check(compare_backups());
+    reset_dir(home_full);
+}
+
+/*
+ * usage --
+ *     Print out the command usage message.
+ */
+static void
+usage(void)
+{
+    fprintf(stderr, "Usage: %s [-w path] [-i number]\n", progname);
+    fprintf(stderr, "\t-w <path> optional path to the wt tool. Default is \"./wt\".\n");
+    fprintf(stderr, "\t-i <number> optional number of iterations to run. Default is 5.\n");
+
+    exit(EXIT_FAILURE);
+}
+
+/*
+ * parse_args --
+ *     Parse command line arguments.
+ */
+static void
+parse_args(int argc, char *argv[])
+{
+    int ch;
+
+    while ((ch = __wt_getopt(progname, argc, argv, "w:i:")) != EOF)
+        switch (ch) {
+        case 'w':
+            wt_tool_path = __wt_optarg;
```

Comment:
If the test knows its location is this necessary?

Reply:
Probably no. I just need to make sure that evergreen works well with `../../wt`.

Reply:
Evergreen patch build failed with `../../wt: not found`. But finished successfully with default path `./wt`. That means that it runs the test from the build folder.

Reply:
Would it be reasonable for the program to try a few well-known locations (with a good comment). Something like "find_wt_path" and it checks for "./wt" which will work for Evergreen, and "../../wt" for running in test/csuite, or "../../../wt" if you're running in test/csuite/wt8659_directory. And then fail if it cannot find one. It would be a pain for either Evergreen or the user to always have to specify a path.

Reply:
OK, good idea. I can do that.